src/main/java/frc/team2412/robot/subsystem/DrivebaseSubsystem.java

```diff
@@ -178,8 +186,21 @@ public DrivebaseSubsystem(SwerveModule fl, SwerveModule fr, SwerveModule bl, Swe
 
         isFieldOrientedEntry = tab.add("Field Oriented", true).getEntry();
 
-        defaultX = gyroscope.getAxis(Pigeon.Axis.ROLL);
-        defaultY = gyroscope.getAxis(Pigeon.Axis.PITCH);
+        tipController = PFFController.ofVector2(TIP_P, TIP_F).setTargetPosition(getGyroscopeXY())
+                .setTargetPositionTolerance(TIP_TOLERANCE);
+
+    }
+
+    public Vector2 getGyroscopeXY() {
+        synchronized (sensorLock) {
+            if (gyroscope instanceof Pigeon)
+                return new Vector2(((Pigeon) gyroscope).getAxis(Pigeon.Axis.ROLL),
+                        ((Pigeon) gyroscope).getAxis(Pigeon.Axis.PITCH));
+            if (gyroscope instanceof NavX)
+                return new Vector2(((NavX) gyroscope).getAxis(NavX.Axis.ROLL),
+                        ((NavX) gyroscope).getAxis(NavX.Axis.PITCH));
```

Comment:
A cleaner way to do this would be to add `getXY()` to the `Gyroscope` class, then implement it in the `NavX` and `Pigeon` subclasses.